kbase-extension/static/kbase/js/common/jobCommChannel.js

```diff
@@ -404,54 +402,6 @@ define([
                     });
                     break;
 
-                /*
-                 * This message must carry all jobs linked to this narrative.
-                 * The job deletion logic, specifically, requires that the job
-                 * actually not exist in the job service.
-                 * NB there is logic in the job management back end to allow
-                 * job notification to be turned off per job -- this would
-                 * be incompatible with the logic here and we should address
-                 * that.
-                 * E.g. if that behavior is allowed, then deletion detection
-                 * would need to move to the back end, since that is the only
-                 * place that would truly know about all jobs for this narrative.
-                 */
-                case 'job_status_all':
-                    /*
-                     * Ensure there is a locally cached copy of each job.
-                     *
-                     */
-                    for (jobId in msgData) {
-                        const jobStateMessage = msgData[jobId];
-                        // cache state data
-                        this.jobStates[jobId] = jobStateMessage.state;
-
-                        this.sendBusMessage(JOB, jobId, RESPONSES.STATUS, {
-                            jobId: jobId,
-                            jobState: jobStateMessage.state,
-                            outputWidgetInfo: jobStateMessage.widget_info,
-                        });
-                    }
-
-                    Object.keys(this.jobStates).forEach((_jobId) => {
-                        if (!msgData[_jobId]) {
-                            // If this job is not found in the incoming list of all
-                            // jobs, then we must both delete it locally, and
-                            // notify any interested parties.
-                            this.sendBusMessage(JOB, _jobId, RESPONSES.STATUS, {
-                                jobId: _jobId,
-                                jobState: {
-                                    job_id: _jobId,
-                                    status: 'does_not_exist',
-                                },
-                            });
```

Comment:
the particularly damaging cruft -- if the backend `job_status_all` call returned `{}`, this would turn off job status listeners for all jobs in `this.jobStates`

Reply:
❗